subprojects/core/src/main/java/org/gradle/initialization/exception/DefaultExceptionAnalyser.java

```diff
@@ -117,4 +128,16 @@ private Throwable findDeepestRootException(Throwable exception) {
             return exception;
         }
     }
+
+    private static @NotNull Throwable patchCircularCause(Throwable current, Throwable parent) {
+        try {
+            Field causeField = Throwable.class.getDeclaredField("cause");
+            causeField.setAccessible(true);
+            current = new Throwable("[CIRCULAR REFERENCE: " + current + "]" );
```

Comment:
I think you can do
```

Throwable replacement = new Throwable("[CIRCULAR REFERENCE: " + current + "]" );
replacement.setStackTrace(current.getStackTrace());
return replacement
```
in order to get closer to reality.

(and avoid messing with the cause field)

Reply:
hmmm actually you need to patch the cause of the parent.
I'd still patch the stacktrace on the replacement though.

Reply:
Good point about copying the stacktrace, I am doing that now.